### PR TITLE
Small cleanup of API code.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Comment by brian-brazil** [comment]
I don't think this makes the code cleaner. Having to bounce around a file to find simple constants is really annoying.

In addition using constants in tests means that we're no longer testing a fixed string but rather some changeable variable, making us more fragile to breaking changes.

**Comment by bwplotka** [comment]
> I don't think this makes the code cleaner. Having to bounce around a file to find simple constants is really annoying.

This might be a subjective opinion. Why do you want to find the value of constant - the constant should be well named in all cases (: Additionally,  it's always better to NOT hardcode important string (many times!), as we might miss some test case to find potential error out (e.g typo in error wrap). To me, it's a cleaner and more reliable code.

> In addition using constants in tests means that we're no longer testing a fixed string but rather some changeable variable, making us more fragile to breaking changes.

The fair point here, I need to agree with. :+1: 

What if I would remove the usage of those constants in test then? :hugs:

**Comment by brian-brazil** [comment]
> Why do you want to find the value of constant - the constant should be well named in all cases (:

So I can tell what the code is actually doing. Every indirection gets in the way of that, and greatly slows down my ability to read and understand code. For every single use of something that looks like a variable I need to spend effort to determine where the variable is defined and what logic is applied to it. Even if if turns out to be a constant in a particular case, I still need to check for shadowing, and add the values to the myriad of other information I need to keep mental track of when reading code.

Code that is self-contained on one line is much easier to understand.

> What if I would remove the usage of those constants in test then? hugs

That wouldn't work for me, this PR seems to be adding indirection for the sake of adding indirection.

**Comment by roidelapluie** [comment]
My preference is also not to have small const like this.

However, invalidParamError is nice and I like it.

**Comment by bwplotka** [comment]
I disagree and commit (: 

Removed constants, leaving the helper only. PTAL :hugs:  (:

**Comment by brian-brazil** [comment]
:+1:

File: web/api/v1/api.go [hunk 1]
```diff
@@ -389,28 +394,23 @@ func (api *API) query(r *http.Request) (result apiFuncResult) {
...
 	end, err := parseTime(r.FormValue("end"))
 	if err != nil {
-		err = errors.Wrapf(err, "invalid parameter 'end'")
-		return apiFuncResult{nil, &apiError{errorBadData, err}, nil, nil}
+		return invalidParamError(err, "end")
 	}
 	if end.Before(start) {
-		err := errors.New("end timestamp must not be before start time")
-		return apiFuncResult{nil, &apiError{errorBadData, err}, nil, nil}
+		return invalidParamError(errors.New("end timestamp must not be before start time"), "end")
```
Comment (roidelapluie) [inline web/api/v1/api.go:404]: I am not sure about this one, it would not render nicely.

File: web/api/v1/api.go [hunk 2]
```diff
@@ -808,20 +807,16 @@ func (api *API) targetMetadata(r *http.Request) apiFuncResult {
...
-
 	var matchers []*labels.Matcher
-
 	var err error
-
 	if matchTarget != "" {
 		matchers, err = parser.ParseMetricSelector(matchTarget)
 		if err != nil {
-			return apiFuncResult{nil, &apiError{errorBadData, err}, nil, nil}
+			return invalidParamError(err, "match[]")
```
Comment (roidelapluie) [inline web/api/v1/api.go:815]: is that `match_target` ?

Comment (bwplotka) [reply web/api/v1/api.go:815]: nicely spotted

Comment (bwplotka) [reply web/api/v1/api.go]: fixed!

**Review by roidelapluie — APPROVED** [review]

**Pull request merged**